Automattic/studio
# Studio: Add UI to import new site during Onboarding and add site modal

Closes https://github.com/Automattic/dotcom-forge/issues/8159
Closes https://github.com/Automattic/dotcom-forge/issues/8158
## Proposed Changes

This PR adds the new import site UI to onboarding screen and to `Add site` modal:
* it adds the functionality to drag and drop the import file
* it adds the functionality to open the file manager and select the file for import

## Testing Instructions

**Testing onboarding screen**

Before starting the testing:
* Add a specific theme to one of your local sites in Studio and make sure it is installed
* Navigate to the Import/Export tab
* Export that site (not the database)

Once you have the backup file:
* Delete all local sites
* Pull the changes from this branch
* Start the app with `STUDIO_IMPORT_EXPORT=true npm start`
* Observe that you can now see a new field on the onboarding screen that says `Import a backup`:

<img width="569" alt="Screenshot 2024-07-25 at 8 22 47 AM" src="https://github.com/user-attachments/assets/4a100594-98fc-4ac1-94d8-15dd664cc3e6">

* Click on `Advanced Settings` button and confirm that you can see the `Local path` appear
* Click again on `Advanced Settings` button  and confirm that the field with `Local path` closes
* Select a backup file to import (that one you previously exported)
* Drag it into the `Add your first site` area (the white part of the onboarding)
* Observe the white overlay and the blue download icon appearing while you are dragging the file
* Observe that once you drag the file, its name is filled into the `Import the backup` field and the trash icon appears:

<img width="532" alt="Screenshot 2024-07-25 at 8 27 57 AM" src="https://github.com/user-attachments/assets/511c6356-a5fe-40aa-9a42-c6436682c906">

* Click on the trash icon and confirm that it clears the input field
* Try dragging an invalid import file e.g. `.png` - confirm that you can't drag it into the form and you see the error
* Select a backup file to import (that one you previously exported)
* Drag it into the `Add your first site` area (the white part of the onboarding)
* This time, proceed with `Add site` button
* Observe that you see the loading screen
* Observe that once the site is created, you get the `Import completed` notification
* Open the new site and navigate to `Themes`
* Confirm that you can see the themes that were present in the export file on the new site
* Confirm that the site works as expected

**Testing with the file manager**

* Delete the created site to get back to the onboarding screen
* Click on the folder icon or on the input field for `Import a backup`
* Observe that it opens the file manager and allows you to select only `tar.gz`, `.zip`, `.sql`
* Select the backup file to import 
* Observe that the input field is populating with the file name
* Proceed with the `Add site` button
* Observe that you see the loading screen
* Observe that once the site is created, you get the `Import completed` notification
* Open the new site and navigate to `Themes`
* Confirm that you can see the themes that were present in the export file on the new site
* Confirm that the site works as expected

**Testing Add site modal**

* Click on the `Add site` button in the sidebar to open the `Add site` modal
* Observe that you can see the new input field to add the backup file on the modal
* Select a backup file to import (that one you previously exported)
* Drag it into the modal area
* Observe that the design with the white overlay and the blue icon appears while you are dragging the file
* Observe that once you drag the file, its name is filled into the `Import the backup` field and the trash icon appears
* This time, proceed with `Add site` button
* Observe that you see the loading screen
* Observe that once the site is created, you get the `Import completed` notification
* Open the new site and navigate to `Themes`
* Confirm that you can see the themes that were present in the export file on the new site
* Confirm that the site works as expected
* Repeat the steps above but instead of dragging the file, test it with the file manager

**Check Edit Site form**

(Added these testing steps as the site form is sharing the same component with `Add site` modal and onboarding

* Navigate to the `Settings` tab for a site that is already created
* Click on `Edit` by the `Site name`
* Observe that you can only see the field to edit the site name and nothing else

**Test feature flag**
* Pull the changes from this branch
* Start the app with `npm start`
* Confirm that you see the UI that is currently available in the app
* Quit the app and start the app with `STUDIO_IMPORT_EXPORT=true npm start`
* Confirm that you see the new UI in the onboarding screen and `Add site` modal
## Pre-merge Checklist
